pkg/kubelet/api/testing/fake_runtime_service.go

```diff
@@ -321,13 +317,13 @@ func (r *FakeRuntimeService) ListContainers(filter *runtimeapi.ContainerFilter)
 	result := make([]*runtimeapi.Container, 0)
 	for _, s := range r.Containers {
 		if filter != nil {
-			if filter.Id != nil && filter.GetId() != s.GetId() {
+			if filter.Id != "" && filter.Id != s.Id {
 				continue
 			}
-			if filter.PodSandboxId != nil && filter.GetPodSandboxId() != s.SandboxID {
+			if filter.PodSandboxId != "" && filter.PodSandboxId != s.SandboxID {
 				continue
 			}
-			if filter.State != nil && filter.GetState() != s.GetState() {
+			if filter.State != runtimeapi.ContainerState_CONTAINER_NOTSPECIFIED && filter.State != s.State {
```

Comment:
Per https://github.com/kubernetes/kubernetes/pull/39158#discussion_r96540658
If we decide remove the default value, we will simplify the code here as well.